framework/src/modules/dpos_v2/module.ts

```diff
@@ -214,4 +235,20 @@ export class DPoSModule extends BaseModule {
 			await snapshotStore.del(key);
 		}
 	}
+
+	private async _updateProductivity(
+		newContext: BlockAfterExecuteContext & { previousTimestamp: number },
+	) {
+		const { logger, previousTimestamp } = newContext;
+
+		const round = new Rounds({ blocksPerRound: this._moduleConfig.roundLength });
+		logger.debug(round, 'Updating delegates productivity for round');
+
+		await updateDelegateProductivity({
```

Comment:
Is there any reason to make auxillary func? if its not being used anywhere else then we can keep the implementation here rather than splitting into two funcs as @shuse2 pointed out

Reply:
done